jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -287,21 +302,40 @@ async def job_script_update(job_script_id: int, job_script: JobScriptUpdateReque
     update_query = (
         job_scripts_table.update()
         .where(job_scripts_table.c.id == job_script_id)
-        .values(job_script.dict(exclude_unset=True))
+        .values(
+            job_script.dict(exclude_unset=True, exclude={"job_script_data_as_string"}),
+        )
         .returning(job_scripts_table)
     )
-    try:
-        result = await database.fetch_one(update_query)
-    except INTEGRITY_CHECK_EXCEPTIONS as e:
-        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
-
-    if result is None:
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail=f"JobScript with id={job_script_id} not found.",
-        )
 
-    return result
+    async with database.transaction():
+
+        try:
+            result = await database.fetch_one(update_query)
+        except INTEGRITY_CHECK_EXCEPTIONS as e:
+            raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
+
+        if result is None:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"JobScript with id={job_script_id} not found.",
+            )
```

Comment:
in terms of the HTTP protocol, as far as I understand, a PUT method should never return 404. In this case the method bust be PATCH. Do you agree? @dusktreader might have some input as well

Reply:
A PUT can definitely return a 404. If you give a valid body but a bad key (the id for the entity to be updated), this should absolutely be a 404.

A 404 should never be returned for a POST.

The only difference between a PUT and a PATCH is that a PUT expects to update an entire existing entity while a PATCH may only partially update an existing entity.

Reply:
I'm not sure if I'm on this page. If there's no key corresponding when the method is PUT then the router would create the key. PATCH would return 404

Reply:
We discussed this. Resolving for now, conversation can carry on separately.